src/components/Topic.jsx

```diff
@@ -0,0 +1,57 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+import axios from "axios";
+
+function Topic() {
+  const { single_topic } = useParams();
+  const [topicList, setTopicList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(true);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  useEffect(() => {
+    axios
+      .get("https://nc-news-pa.herokuapp.com/api/articles")
+      .then((res) => {
+        const filteredTopic = res.data.articles.filter(
+          (article) => article.topic === single_topic
+        );
```

Comment:
you should *not* be using a javascript filter here

not only is it less efficient and more computationally expensive it can also lead to imcomplete data!

In real world apps data will likely be paginated (sent page-by-page). If you were to just filter page 1 data you could end up seeing no data at all!

You should be using your API to get articles by topic

Reply:
Hi doug. There's no topic endpoint  with articles in my Api. there's only a list of topics. any suggestions on how i can get the articles?